client/src/componenets/AllCharacters.js

```diff
@@ -1,8 +1,40 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+import Character from './Character';
+import CharacterModel from '../models/CharacterModel'
+import Styles from './AllCharacters.module.css'
+
+import useAxios from '../hooks/useAxios';
```

Comment:
Dont use in the useAxios like that
Please change to another structure.
All the logic placed on service by feature at the code we dont want to see all the API call.